[Event Hubs] Updates logging to use core logging package

chradek opened the pull request:
This builds on #4926 
Fixes #2845 

Event Hubs currently creates debugging namespaces for nearly every class. These changes bring us in-line with the Azure SDK guidelines by consolidating namespaces to the package name with 4 log levels.

I also pulled in the core-logging package that @bterlson worked on to get support for setting log levels. The 1st commit cherry-picking Brian's core-logging from his PR with some minor changes to get it to build in the monorepo's current state.

Edit: I did not remove any of our existing logs. While we should consider if our logging is too verbose, these changes update the existing logging to use the new `azure:event-hubs:[log-level]` namespaces while taking into consideration the SDK guidelines for which log levels to use.

### ToDo:
- [ ] update README troubleshooting section

xirzec commented:
Did you end up having to change Brian's original PR or should we just flip that out of draft and merge it as is? Combining that PR with this one is making this one feel super heavy.

chradek commented:
I had to make some changes so that it would build in browsers and I fixed some minor things around extract-api (the api review md and the rolled-up types), but they were pretty small. Brian's PR also updated Identity with logging...but I don't mind making changes to his PR instead if we'd rather do that.

chradek commented:
Alternatively I could remove the event hubs-specific things from this PR to make it just core-logging. Then once that's merged resubmit the event-hubs bits.

ramya-rao-a commented:
#4926 also deals with changes to core-http and others, so we can't wait for that to be merged.
I would suggest to create a PR that only adds the logger package first. We can put a rush on it tomorrow morning, and then another PR to absorb those changes to Event Hubs. @bterlson can then add a PR to add logging to core-http if there is time tomorrow.

xirzec commented:
I agree with @ramya-rao-a and would be happy to help shepherd this in tomorrow morning. 🐑

chradek commented:
Closing in favor of #5401
